File: Common/InMotionGIT.Common - Copy/DataAcceess/WhereStatements/BooleanCondition.vb
Namespace DataAccess.WhereStatements

    Public Class BooleanCondition(Of t)

        Private ReadOnly fieldName As String
        Private ReadOnly tableType As IWhereStatement

        Public Sub New(name As String, base As IWhereStatement)
            fieldName = name
            tableType = base
        End Sub

        Public Function IsTrue() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

        Public Function NotIsTrue() As LogicalOperator(Of t)
            tableType.command &= String.Format("{0} IS NOT NULL", fieldName)
            Return New LogicalOperator(Of t)(tableType)
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/DataAcceess/WhereStatements/BooleanCondition.cs

namespace InMotionGIT.Common.DataAccess.WhereStatements
{

    public class BooleanCondition<t>
    {

        private readonly string fieldName;
        private readonly IWhereStatement tableType;

        public BooleanCondition(string name, IWhereStatement @base)
        {
            fieldName = name;
            tableType = @base;
        }

        public LogicalOperator<t> IsTrue()
        {
            tableType.command += string.Format("{0} IS NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

        public LogicalOperator<t> NotIsTrue()
        {
            tableType.command += string.Format("{0} IS NOT NULL", fieldName);
            return new LogicalOperator<t>(tableType);
        }

    }

}